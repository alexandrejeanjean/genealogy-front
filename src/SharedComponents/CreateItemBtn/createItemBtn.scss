.new-item-btn .new-card {
  width: 20em;
  height: 20em;
  margin: 0;
  img {
    width: 3.5em;
  }
}
