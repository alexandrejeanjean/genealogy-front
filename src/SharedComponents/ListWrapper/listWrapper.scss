.list-item-wrapper {
  margin-top: 2em;
  .list-items {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
  }
}
