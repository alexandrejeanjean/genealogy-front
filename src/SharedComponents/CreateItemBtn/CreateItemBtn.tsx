import React from "react";
import { add } from "../../assets/imgPath";
import "./createItemBtn.scss";

type TChild = {
  handleClick: () => any;
};

const CreateItemBtn = ({ handleClick }: TChild) => {
  return (
    <div className="card-wrapper new-card">
      <button className="card" onClick={handleClick}>
        <img src={add} alt="ajouter" />
      </button>
    </div>
  );
};

export default CreateItemBtn;
