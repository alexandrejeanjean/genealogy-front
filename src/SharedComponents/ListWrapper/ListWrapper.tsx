import React from "react";
import { Row, Col } from "react-bootstrap";
import Card from "../Card/Card";
import "./listWrapper.scss";

type TListWrapper = {
  datas: any;
  children?: any;
  getSubText?: (id: number) => string;
  getAvatar?: (id: number) => any;
  handleClick: (id: number) => void;
  handleDelete: (item: any) => void;
};

const ListWrapper = ({
  datas,
  children,
  getAvatar,
  getSubText,
  handleClick,
  handleDelete,
}: TListWrapper) => {
  return (
    <Row>
      <Col className="list-item-wrapper">
        <ul className="list-items">
          {datas &&
            datas.map((item: any, i: number) => (
              <li key={i + `_person`}>
                <Card
                  picture={getAvatar && getAvatar(item.roleId)}
                  text={item.name || item.firstname + " " + item.lastname}
                  subtext={getSubText && getSubText(item.roleId)}
                  handleClick={() => handleClick(item.id)}
                  handleDelete={() => handleDelete(item)}
                />
              </li>
            ))}
          <li>{children}</li>
        </ul>
      </Col>
    </Row>
  );
};

export default ListWrapper;
