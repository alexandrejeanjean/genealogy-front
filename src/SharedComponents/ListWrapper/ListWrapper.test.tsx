import React from "react";
import { shallow, mount } from "enzyme";
import ListWrapper from "./ListWrapper";
import CreateItemBtn from "../CreateItemBtn/CreateItemBtn";

const setModalShow = jest.fn();
const BtnProps = {
  handleClick: setModalShow,
};

const Children = <CreateItemBtn {...BtnProps} />;

const ListProps = {
  datas: [{ name: "Armstrong" }],
  children: Children,
  getSubText: jest.fn(),
  handleClick: jest.fn(),
  handleDelete: jest.fn(),
};

describe("ListWrapper", () => {
  it("should render correctly", () => {
    const Component = shallow(<ListWrapper {...ListProps} />);
    expect(Component).toMatchSnapshot();
  });

  it("should render the right number of childrens depending on length of datas array passed in props +1 delete button", () => {
    const Component = mount(<ListWrapper {...ListProps} />);
    expect(Component.find(".card-wrapper").length).toBe(
      ListProps.datas.length + 1
    );
    Component.unmount();
  });
});
