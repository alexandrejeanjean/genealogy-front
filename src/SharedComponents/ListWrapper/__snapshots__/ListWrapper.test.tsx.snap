// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListWrapper should render correctly 1`] = `
<Row
  noGutters={false}
>
  <Col
    className="list-item-wrapper"
  >
    <ul
      className="list-items"
    >
      <li
        key="0_person"
      >
        <Card
          handleClick={[Function]}
          handleDelete={[Function]}
          text="Armstrong"
        />
      </li>
      <li>
        <CreateItemBtn
          handleClick={[MockFunction]}
        />
      </li>
    </ul>
  </Col>
</Row>
`;
